src/main/java/br/com/deveficiente/bolaoapi/services/championship/Championship.java

```diff
@@ -54,4 +61,19 @@ public Championship(String name, LocalDate startDate, Integer totalTeams, Set<Te
         this.totalTeams = totalTeams;
         this.teams.addAll(teams);
     }
+
+    public void addMatch(@NotNull Match match) {
+        Assert.isTrue(hasNotMatchInRound(match), "A team cannot play 2 matches in same round.");
+
+        this.matches.add(match);
+    }
+
+    public boolean hasNotMatchInRound(Match match) {
+        Optional<Match> overlay = matches.stream()
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Não que mude muita coisa, mas justamente para este caso tem o método anyMatch :).

Reply:
Hmm é msm, bem mais expressivo!